open-metadata-implementation/adapters/open-connectors/rest-client-connectors/spring-rest-client-connector/src/main/java/org/odpi/openmetadata/adapters/connectors/restclients/spring/SpringRESTClientConnector.java

```diff
@@ -41,49 +49,317 @@
+        final String methodName = "SpringRESTClientConnector";
+        /* TODO: Disable SSL cert verification -- for now */
 
+        // Numerous opportunities for error. We'll propogate the base exception to caller, and it will appear in audit log
+        try {
+            SSLContext sc = SSLContext.getInstance(protocol);
 
-        /* TODO: Disable SSL cert verification -- for now */
-        HttpsURLConnection.setDefaultHostnameVerifier(bypassVerifier);
-        SSLContext sc = SSLContext.getInstance("SSL");
-        sc.init(null, INSECURE_MANAGER, null);
-        HttpsURLConnection.setDefaultSSLSocketFactory(sc.getSocketFactory());
+            TrustManager[] trustManagers = null;
+            KeyManager[] keyManagers = null;
+
+            // If verification is to be done we need to configure the truststore & keystore
+            if (sslVerify) {
+
+                // First configure the truststore -- this is always needed for SSL (one way) & is where certs/CA we trust sit
+                InputStream tsFile;
+
+                // Truststore -- where we store CAs & certificates we trust. Used by client in 1way and mutual SSL
+                KeyStore tStore = KeyStore.getInstance("PKCS12"); // only support the open pkcs12 format
+                TrustManagerFactory tmFactory = TrustManagerFactory.getInstance("X509");
+
+                // Allow use of classpath - to find resource inside jar. Useful for out of the box config
+                if (StringUtils.startsWithIgnoreCase(truststore, "classpath:")) {
+                    tsFile = this.getClass().getClassLoader().getResourceAsStream(StringUtils.removeStart(truststore, "classpath:"));
+                } else
+                    // otherwise absolute filename
+                    //TODO: If truststore is null - or file not found, we'll hit an exception here
+                    tsFile = new FileInputStream(truststore);
+
+                // Load the keystore, initialise the factory
+                tStore.load(tsFile, truststorePassword.toCharArray());
+                tmFactory.init(tStore);
+                tsFile.close();
+
+                // Get the trust manager list for our ssl context initialization
+                trustManagers = tmFactory.getTrustManagers();
+
+                // Next we configure the keystore - but only if a keystore is configured - for mutual SSL
+                if (StringUtils.isNotBlank(keystore)) {
+                    InputStream ksFile;
+
+                    KeyStore kStore = null;
+                    KeyManagerFactory kmFactory = null;
+                    if (StringUtils.isNotBlank(keystore)) {
 
-        restTemplate = new RestTemplate();
+                        //TODO: Default KeyManagers chooses a cert to use ie by alpha sort -- Implement own X509KeyManager if required to choose by alias - the default
+                        kStore = KeyStore.getInstance("PKCS12");
```

Comment:
why to have such a limitation?

Reply:
pkcs12 is standard across environments, and the default from java 9 & above. JKS is java specific
For the intial implementation I simply didn't spend the time to allow jks, but this would be easy if needed. 
Do you use jks or pkcs12?